esx_service/utils/auth_api.py

```diff
@@ -488,9 +488,28 @@ def vm_in_any_tenant(vms):
 
     return None
 
+
+def named_tenant(func):
```

Comment:
I'd suggest to rename to "check_tenant_name", this is just my personal preference though.